### PR TITLE
Support for Jest generated reports

### PR DESCRIPTION
**Pull request by jacekk**
I've used this tool in a React project in which [Jest](https://jestjs.io/) is being used for code coverage reporting. There are, sadly, two problems with the `main` branch. Thus, introduced some improvements. Problems:

**A)**
`file` entry in `clover.xml` contains `name` and `path` attributes. `name` is just the file name with extension. While `path` contains the full file system path. Example:

```xml
<file name="file-one-A.js" path="/home/foo/bar-project/src/dir-one/file-one-A.js">
```

**B)**
The `xml2json` package is not returning a list of files, when only a single spec is within given directory. Namely, not this:

```ts
package?: { file: CloverFileXML[] }[];
```

but this:

```ts
package?: { file: CloverFileXML | CloverFileXML[] }[];
```

seem to be the right type.

----

Both these issues have been fixed in this PR. Reproduced and tested in the following changsets, which are identical when it comes to modified `spec` files. The difference is in workflows; negligible change.

**Broken:**

https://github.com/jacekk/comment-coverage-clover--jest-support/pull/1

image

---

**Fixed:**

https://github.com/jacekk/comment-coverage-clover--jest-support/pull/2

image

---

**In summary:**

- Files directories were visible.
- Spec file from a directory with a single spec file were not listed.

Additionally, `filters === []` seems to be always `false` - JS is reference-based for objects and arrays.

### CODE REVIEW

File: src/clover/index.ts [hunk 1]
```diff
@@ -61,28 +64,34 @@ export const fromString = (str: string): Stats => {
       branchs: new Coverage(m.conditionals, m.coveredconditionals),
     },
     allFiles
```
Comment (lucassabreu) [inline src/clover/index.ts:66]: just to reduce the changes on the file, you could do this:
```suggestion
    allFiles
      .map(f => {
        f._attributes.name = file._attributes.path || file._attributes.name
        return f
      })
```

doing that the other changes of this map-reduce can be reverted.

Comment (jacekk) [reply src/clover/index.ts:66]: @lucassabreu Improved in https://github.com/lucassabreu/comment-coverage-clover/pull/27/commits/4d89722e7978febead5047ddbf4d1c6c9edce8b4

**Review by lucassabreu — APPROVED** [review]

**Comment by lucassabreu** [comment]
thank you for the contribution @jacekk
